src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EvpPkey.Rsa.cs

```diff
@@ -62,6 +62,41 @@ internal static SafeEvpPKeyHandle RsaGenerateKey(int keySize)
             return written;
         }
 
+        [DllImport(Libraries.CryptoNative)]
+        private static extern int CryptoNative_RsaEncrypt(
+            SafeEvpPKeyHandle pkey,
+            ref byte source,
+            int sourceLength,
+            RSAEncryptionPaddingMode paddingMode,
+            IntPtr digestAlgorithm,
+            ref byte destination,
+            int destinationLength);
+
+        internal static int RsaEncrypt(
+            SafeEvpPKeyHandle pkey,
+            ReadOnlySpan<byte> source,
+            RSAEncryptionPaddingMode paddingMode,
+            IntPtr digestAlgorithm,
+            Span<byte> destination)
+        {
+            int written = CryptoNative_RsaEncrypt(
+                pkey,
+                ref MemoryMarshal.GetReference(source),
+                source.Length,
+                paddingMode,
+                digestAlgorithm,
+                ref MemoryMarshal.GetReference(destination),
+                destination.Length);
+
+            if (written < 0)
+            {
+                Debug.Assert(written == -1);
```

Comment:
do you need to zero memory on failure here or below?

Reply:
There isn't any secret data being exposed or left around.  And, unlike GCM decryption with integrity failures, there isn't any bad behavior from the user depending on partial data being written.